src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,144 @@
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { TodosContext } from '../../utils/TodosContext';
+import { deleteTodo, updateTodo } from '../../api/todos';
+import { ErrorContext } from '../../utils/ErrorContextProvider';
+import { Errors } from '../../types/Errors';
+
+type Props = Todo & { isLoading: boolean };
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  title,
+  completed,
+  isLoading,
+  id,
+}) => {
+  const [currentTitle, setCurrentTitle] = useState(title);
+  const [showModal, setShowModal] = useState(isLoading);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const { setTodos } = useContext(TodosContext);
+  const { showError } = useContext(ErrorContext);
+
+  const todoInput = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing && todoInput.current) {
+      todoInput.current.focus();
+    }
+  }, [isEditing]);
+
+  useEffect(() => setShowModal(isLoading), [isLoading]);
```

Comment:
redundant `useEffect`
`isLoading` is a prop, so once its value changes component will be re-rendered